### PR TITLE
CircleCI config tweaks

### PR DESCRIPTION
**Pull request by deanwhillier**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -229,10 +227,16 @@ jobs:
       - checkout
       - attach_workspace:
           at: ./dist
+      - run:
+          name: "Setup files for aws-s3"
+          command: |
+            mkdir -p ./aws-s3-dist
+            cp ./dist/{macos-release,win-release,linux}/* ./aws-s3-dist
```
Comment (deanwhillier) [inline .circleci/config.yml]: Flattens the file structure for aws upload.

File: .circleci/config.yml [hunk 2]
```diff
@@ -249,12 +253,17 @@ jobs:
...
+              -u ${CIRCLE_PROJECT_USERNAME} \
+              -draft \
               --body="$(./scripts/generate_release_markdown.sh $VERSION)" \
               --name="${RELEASE_TITLE}" $( [[ $VERSION =~ "-rc" ]] && printf %s "-prerelease") \
-              -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} \
-              -delete ${VERSION} ./ghr-dist
+              -r ${CIRCLE_PROJECT_REPONAME} \
+              -c ${CIRCLE_SHA1} \
+              -delete \
+              v${VERSION} ./ghr-dist
```
Comment (deanwhillier) [inline .circleci/config.yml]: added `v` at the beginning to ensure the draft tag has format v#.#.#

**Comment by Willyfrog** [comment]
I tend to forget that bash+circle is not a very forgiving environment (which in its use case is a good thing)

**Comment by deanwhillier** [comment]
@Willyfrog, no worries!

Question: do you know why the aws-s3 upload task is being triggered here in master? I'm not seeing it ...

image

**Comment by Willyfrog** [comment]
is the branch tagged? that is the filter it has

**Comment by deanwhillier** [comment]
Ya, exactly. No tag and only 1 of 3 required jobs 🤷‍♂️

**Comment by deanwhillier** [comment]
Maybe something here: https://circleci.com/docs/2.0/configuration-reference/#tags

> Additionally, if a job requires any other jobs (directly or indirectly), you must specify tag filters for those jobs

**Comment by deanwhillier** [comment]
aaaaannnnnddd, maybe not ...

**Comment by Willyfrog** [comment]
Removing from 4.4 as it is not ready

**Pull request merged**